src/jit/instrsxarch.h

```diff
@@ -474,8 +474,8 @@ INST3( vpbroadcastb, "pbroadcastb" , 0, IUM_WR, 0, 0, BAD_CODE,     BAD_CODE, SS
 INST3( vpbroadcastw, "pbroadcastw" , 0, IUM_WR, 0, 0, BAD_CODE,     BAD_CODE, SSE38(0x79))   // Broadcast int16 value from reg/memory to entire ymm register
 INST3( vpbroadcastd, "pbroadcastd" , 0, IUM_WR, 0, 0, BAD_CODE,     BAD_CODE, SSE38(0x58))   // Broadcast int32 value from reg/memory to entire ymm register
 INST3( vpbroadcastq, "pbroadcastq" , 0, IUM_WR, 0, 0, BAD_CODE,     BAD_CODE, SSE38(0x59))   // Broadcast int64 value from reg/memory to entire ymm register
-INST3( vextractf128, "extractf128" , 0, IUM_WR, 0, 0, SSE3A(0x19),  BAD_CODE, BAD_CODE)      // Extract 128-bit packed floating point values
-INST3( vextracti128, "extracti128" , 0, IUM_WR, 0, 0, SSE3A(0x39),  BAD_CODE, BAD_CODE)      // Extract 128-bit packed integer values
+INST3( vextractf128, "extractf128" , 0, IUM_WR, 0, 0, SSE3A(0x19),  BAD_CODE, SSE3A(0x19))   // Extract 128-bit packed floating point values
+INST3( vextracti128, "extracti128" , 0, IUM_WR, 0, 0, SSE3A(0x39),  BAD_CODE, SSE3A(0x39))   // Extract 128-bit packed integer values
```

Comment:
Fixed a mistake from https://github.com/dotnet/coreclr/pull/16957, looks like an oversight.

Reply:
No, this was explicit, there is no rm encoding for vextractf128 or vextracti128

Reply:
Tests failed on that changes. `emitIns_R_AR_I` seems to require `RM` field.

Reply:
That is because you should be calling `emitIns_AR_R_I`, which is different.

Reply:
We do have a `emitIns_AR_R_I` now...

Reply:
We do not have a `emitIns_AR_R_I` now...

Reply:
We can use `emitIns_R_AR_I` with swapped operands
```cpp
emit->emitIns_R_AR_I(ins, attr, op2Reg, op1Reg, 0, (int)i);
```
It seems not worthwhile to add a `emitIns_AR_R_I` for the one intrinsic.

Reply:
No. That was one of the things I fixed in #16957. We had multiple GCStress failures because things were "just swapped".

Reply:
> We had multiple GCStress failures because things were "just swapped".

You did not change the "swap" for `Avx/Avx2.ExtractVector128`, you just changed `Sse41.Extract`.

Reply:
> You did not change the "swap" for Avx/Avx2.ExtractVector128, you just changed Sse41.Extract.

Yes, it looks like I missed that they were also swapped for `ExtractVector128`.

In any case, the proper fix should be to:
* Add and plumb through `emitIns_AR_R_I`
* Update `NI_AVX_ExtractVector128` and `NI_AVX2_ExtractVector128` to call `emit->emitIns_R_AR_I(ins, attr, op1Reg, op2Reg, 0, (int)i);`

Reply:
Perhaps @CarolEidt could comment if I'm missing something here, or if there is a potentially better fix.

Reply:
> We had multiple GCStress failures because things were "just swapped".

Question. Why does swapping the operands in the emitter impacts GC? I think the final codegen will be same as `emitIns_AR_R_I `.

Reply:
> Why does swapping the operands in the emitter impacts GC?

The emitter takes care of updating the GC info, and it bases that updating on what it understands is the target register (if any). I'm not absolutely certain it matters in the `AR` case, as the pointer value doesn't change. I haven't thought this through deeply, so take this all with a grain of salt.

That said, we have had numerous issues with "just swapping operands" in the past, so I *really* don't want to go down that path.

@tannergooding when you said:
> - Add and plumb through emitIns_AR_R_I
> - Update NI_AVX_ExtractVector128 and NI_AVX2_ExtractVector128 to call emit->emitIns_R_AR_I(ins, attr, op1Reg, op2Reg, 0, (int)i);

Did you mean `emitIns_AR_R_I` in the second bullet?

Reply:
> Did you mean emitIns_AR_R_I in the second bullet?

Yes.

Reply:
> The emitter takes care of updating the GC info, and it bases that updating on what it understands is the target register (if any). 

Thanks for explaining. So we cannot swap target and operand register, but can we swap two operand registers (unrelated question)?

>  we have had numerous issues with "just swapping operands" in the past, so I really don't want to go down that path.

Can we merge this PR at first and add `emitIns_AR_R_I ` in the next PR?

Reply:
I don't think we should, the code is likely broken in the same manner as the code I just fixed in #16957

Reply:
> the code is likely broken in the same manner as the code I just fixed 

You did not fix `Avx/Avx2.ExtractVector128`, the changes (removing the `RM` field of `extractf128` and `extracti128`) from #16957 makes an assertion failure (`emitIns_R_AR_I` requires `RM` encoding).

Reply:
BTW, I tested this PR with `GCStress=8/4/3/2`, `ExtractVector128` tests all passed.

Reply:
> You did not fix Avx/Avx2.ExtractVector128, the changes (removing the RM field of extractf128 and extracti128) from #16957 makes an assertion failure (emitIns_R_AR_I requires RM encoding).

Right, I missed that `ExtractVector128` was also swapping operands and needed to be changed as well. However, I did fix other code that was broken because it was incorrectly trying to be R_RM, when it is actually RM_R (`NI_SSE41_Extract`, for example).

The overall point is that:
* The code, without any changes, is currently wrong (because I missed something).
* The new changes are also ultimately incorrect (because it is not handling something that is known to cause failures in similar code)

As such, it is my opinion that the code should be fixed now rather than as a later PR.

I will, however, ultimately defer to @CarolEidt.

Reply:
> The new changes are also ultimately incorrect (because it is not handling something that is known to cause failures in similar code)

Question. How can I test the issue? Did you get `ExtractVector128` failures from `GCStress` tests?

Reply:
Before #16957:
* Code was bad, caused GCStress failures
  * When run locally with `EnableIncompleteISAClass`

After #16957:
* Code is still bad, but now causes an assert for `EnableIncompleteISAClass`
  * This was missed because only GCStress and RegStress jobs were run in the final set of changes

After #17030 (this PR):
* I would like to see this passing for regular tests, GCStress tests, and RegStress tests

Reply:
> Question. How can I test the issue?

You have to run locally with both `EnableIncompleteISAClass` and the appropriate Jit Stress flags (such as `COMPlus_GCStress=0xC`)

Reply:
> COMPlus_GCStress=0xC

Yes, I tried all the `GCStress` values and  `ExtractVector128` tests all passed...

> As such, it is my opinion that the code should be fixed now rather than as a later PR.

Let me try to add a `emitIns_AR_R_I`.